src/component/FetchItems.js

```diff
@@ -21,32 +24,38 @@ const FetchItems = ({ token, setToken }) => {
   const concatPath = `/lists/${uniqueToken}/items`;
 
   return (
-    //fetches specific collection, refreshes every time database chages
-    <FirestoreCollection
-      //name of collection you want to collect, with filter you can narrow this down to specific document
-      path={concatPath}
-      // optional more parameters (sort, limit, filter)
-      render={({ isLoading, data }) => {
-        // if data is taking too long you can redirect or alert user it's still loading (i think)
-        return isLoading ? (
-          <div>still loading</div>
-        ) : (
-          <div>
-            <h2>Items</h2>
-            <ul>
-              {data.map(item => (
-                <li key={item.id}>
-                  <div className={item.name}>{item.name}</div>
-                </li>
-              ))}
-            </ul>
-            <DeleteToken token={token} setToken={setToken} />
-            <Navbar />
-          </div>
-        );
-      }}
-    />
+    <React.Fragment>
+      <FirestoreCollection
+        //name of collection you want to collect, with filter you can narrow this down to specific document
+        path={concatPath}
+        // optional more parameters (sort, limit, filter)
+        render={({ isLoading, data }) => {
+          // Renders according to whether or not the list is empty
+          if (isLoading) {
+            return <div>Still Loading...</div>
+          } else if (empty) {
+            return <Link className="emptyAddItemButton" to="/add">Add your first Item</Link>
+          } else {
+            return (
+              <div>
+                <h2>Items</h2>
+                <ul>
+                  {data.map(item => (
+                    <li key={item.id}>
+                      <div className={item.name}>{item.name}</div>
+                    </li>
+                  ))}
+                </ul>
+              </div>
+            )
+          }
+        }}
+        
+        />
+      <DeleteToken token={token} setToken={setToken} />
+      <Navbar />
+    </React.Fragment>
   );
 };
 
-export default FetchItems;
+export default withFirestore(FetchItems);
```

Comment:
Good usage of HoCs :)

Reply:
Thanks! But what's an HoC? 

![bitmoji](https://render.bitstrips.com/v2/cpanel/9d438724-a1e5-44b9-9172-377503bbffe6-141935a4-509e-461c-bba3-4228c119c4d3-v1.png?transparent=1&palette=1&width=100)

Reply:
Found [it!](https://reactjs.org/docs/higher-order-components.html)

Reply:
Thank you @gunderodd!